src/main/arm/createUiDefinition.json

```diff
@@ -168,12 +168,22 @@
                                 "Standard_B1ls",
                                 "Standard_A0",
                                 "Basic_A0",
-                                "Standard_B1s"
+                                "Standard_B1s",
+                                ${azure.armBased.vmSize.list}
                             ]
                         },
                         "osPlatform": "Linux",
                         "count": "1"
                     },
+                    {
+                        "name": "invalidVMSizeInfo",
+                            "type": "Microsoft.Common.InfoBox",
+                            "visible": "[contains(steps('SingleServerConfig').vmSizeSelect,'p')]",
+                            "options": {
+                                "icon": "Error",
+                                "text": "The VM size you selected includes the feature letter 'p', indicating it uses ARM CPUs. ARM platform is not supported. Please select a different VM size. For more information, refer to the <a href='https://learn.microsoft.com/azure/virtual-machines/vm-naming-conventions' target='_blank'>Azure virtual machine sizes naming conventions</a>."
+                        }
+                    },
```

Comment:
This is good. Can you please assure me that this validation feature will be applied across all of our partner vendor's offers?

Reply:
Yes, @galiacheng is working on wls offres, @backwind1233 is working on jboss eap offers, and I'm woking on twas/liberty offers.